libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1375,6 +1375,24 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			setLValue<MemoryItem>(_memberAccess, *_memberAccess.annotation().type);
 			break;
 		}
+		case DataLocation::CallData:
+		{
+			solUnimplementedAssert(!type.isDynamicallyEncoded(), "");
+			m_context << type.calldataOffsetOfMember(member) << Instruction::ADD;
+			// For non-value types the calldata offset is returned directly.
+			if (_memberAccess.annotation().type->isValueType())
+			{
+				solAssert(_memberAccess.annotation().type->calldataEncodedSize(false) > 0, "");
+				CompilerUtils(m_context).loadFromMemoryDynamic(*_memberAccess.annotation().type, true, false, false);
+			}
+			else
+				solUnimplementedAssert(
```

Comment:
```suggestion
				solAssert(
```
Are you planning on implemented anything else apart from arrays and structs? :)

Reply:
This can only happen for something that doesn't exist yet anyways and if we add something that could fail here, then handling that is unimplemented, right :-)? At least that was the idea - I can change to an assert, though, I don't mind either :-).